### DISCUSSION TITLE
sync between weblate and PRs created in the repo?

### DISCUSSION THREAD

**Discussion by tessus** [WeblateOrg/weblate#8953] [Q&A]
We are currently using PRs to update `.po` files in our project [Joplin](https://github.com/laurent22/joplin) and I was thinking about leveraging Weblate to make it easier for people who don't use git.

However, I was wondering, if it would still be possible to allow tech-savvy people to submit PRs to update the translations. Or do we have to require all people to use Weblate instead?

As far as I could ascertain from the docs, the repo will be updated when there is an update to a translation in Weblate, but I couldn't find anything about the opposite direction.

**Comment by nijel** [comment]
Making changes in the repository is fine for Weblate. Just try to avoid conflicts – beware of people changing the same section of PO files on both sides.

**Reply by tessus** [reply]
Thanks for the reply.

> Making changes in the repository is fine for Weblate.

So weblate will also be updated automatically? I couldn't find a guide that explains how to set it up so that the update happens in both directions, or maybe I didn't look hard enough.
Also, I am not sure what user we should use to setup weblate. The joplinbot user or shall we use one of our "personal" gh users? I'll do some more digging.

> Just try to avoid conflicts 

Yep, this makes sense. However, in certain situations avoiding a conflict might not be under our purview. Although the person who merges the PRs has somewhat some control.

I guess if there is a conflict there is still a way to resolve the conflict. e.g. revert PR, rebase PR on updated translations from Weblate, or do manual conflict resolution, which will most likely be a pain.

**Reply by nijel** [reply]
The integration is described in the documentation: https://docs.weblate.org/en/latest/admin/continuous.html
